### PR TITLE
Feature/api start

### PR DESCRIPTION
**Pull request by RoaldL**
- Implement http.api (POST) for sos_model_runs/xxx/start
- Implement http.api (GET) for sos_model_runs/xxx/status
- Redesign http.api tests using a scheduler and datafileinterface mock
- Design controller tests using Popen mock
- Restructure the http.api and scheduler to `https://github.com/nismod/smif/wiki/code_style`

### CODE REVIEW

**Comment by RoaldL** [comment]
@tomalrussell - Tests are passing now, ready for review

File: src/smif/http_api/crud.py [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+
+class SosModelRunAPI(MethodView):
+    """Implement CRUD operations for sos_model_run configuration data
+    """
+    def get(self, sos_model_run_name=None, action=None):
+        """Get sos_model_runs
+        all: GET /api/v1/sos_model_runs/
+        one: GET /api/vi/sos_model_runs/name
+        """
+        # return str(current_app.config)
```
Comment (tomalrussell) [inline src/smif/http_api/crud.py]: Forgot to remove?

File: src/smif/controller/scheduler.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
+""" Implements methods for loading, building and executing model runs
+and a Scheduler that runs instances of smif as a subprocess
+"""
+from subprocess import PIPE, Popen
+
+
+class Scheduler(object):
```
Comment (tomalrussell) [inline src/smif/controller/scheduler.py]: Add a class docstring

File: src/smif/controller/scheduler.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        ----------
+        model_run_name: str
+            Name of the modelrun
+        args: dict
+            Arguments for the command-line interface
+
+        Exception
+        ---------
+        Exception
+            When the modelrun was already started
```
Comment (tomalrussell) [inline src/smif/controller/scheduler.py]: Should this also currently fail here if ANY modelrun has already started?

File: src/smif/app/README.md [hunk 4]
```diff
@@ -16,12 +16,12 @@ npm run build  # runs build script specified in package.json
...
-To develop against the python HTTP API, run the API server against test fixtures
-in another terminal with:
+To develop against the python HTTP API, just run the smif app.
+Enable more detailed debug information by activating the Flask
+debug flag and smif verbosity:
 
 ```bash
-cd ../http_api/
-FLASK_APP=smif.http_api.app FLASK_DEBUG=1 flask run
+FLASK_DEBUG=1 smif -v app
```
[the remark this answers is not in the document]
Comment (RoaldL) [reply src/smif/app/README.md]: Done

File: src/smif/http_api/crud.py (hunk 1 above)
Comment (RoaldL) [reply src/smif/http_api/crud.py]: Done

File: src/smif/controller/scheduler.py (hunk 2 above)
Comment (RoaldL) [reply src/smif/controller/scheduler.py]: Done

File: src/smif/controller/scheduler.py (hunk 3 above)
Comment (RoaldL) [reply src/smif/controller/scheduler.py]: No it is possible to run models concurrently. As discussed we will leave this like this now.

**Pull request merged**